src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -125,6 +125,14 @@ namespace winrt::TerminalApp::implementation
             p.SetActionMap(_settings.ActionMap());
         }
 
+        if (_extensionPalette)
+        {
+            // the extension palette had been loaded with the previous settings
+            // reload it with the new settings
+            _extensionPalette = nullptr;
+            _loadQueryExtension();
```

Comment:
what happens if the user changes the settings (e.g. just touches `settings.json`) while the palette is open?

Reply:
Thanks for the catch, fixed!